src/CMakeLists.txt

```diff
@@ -30,6 +30,14 @@ if(MAPMATCHING_AVAILABLE)
     add_definitions(-DMAPMATCHING_AVAILABLE)
 endif()
 
+if(S2INCLUDES)
+    include_directories(${S2INCLUDES})
+endif()
+
+if(S2LIBS)
+    link_directories(${S2LIBS})
+endif()
+
```

Comment:
Do we need to move S2 section here? I am mainly wondering and trying to understand the reason behind it.

Reply:
Without that move, the link directory is not passed to the compiler resulting in the linker not finding the lib. It seems CMake only considers link directories defined before creating the executable.

Reply:
Thanks! Resolving and merging then